server/src/database/taxa.rs

```diff
@@ -161,11 +149,18 @@ impl TaxaProvider {
     pub async fn species(&self, filters: &Vec<SpeciesFilter>, page: i64, per_page: i64) -> PageResult<Species> {
         use schema_gnl::species::dsl::*;
         let mut conn = self.pool.get().await?;
+        let species_filters = with_species_filters(&filters);
 
-        let records = species
+        let mut query = species
             .filter(status.eq_any(ACCEPTED_NAMES))
             .filter(rank.eq_any(SPECIES_RANKS))
-            .filter(with_species_filters(&filters).unwrap())
+            .into_boxed();
+
+        if let Some(filter) = species_filters {
+            query = query.filter(filter);
+        }
```

Comment:
this is a good change to eliminate the unwrap :raised_hands: 
at a glance the only unknown for me is whether we want it to silently ignore the filter if it fails to convert or if it should return an error